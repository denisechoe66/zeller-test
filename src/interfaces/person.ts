export interface Person {
  firstName: string;
  lastName: string;
  role: string;
}
