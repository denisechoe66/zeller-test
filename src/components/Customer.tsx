import React from 'react';
import { Person } from '../interfaces/person';
import styled from 'styled-components';

interface CustomerProps {
  customer: Person;
}

const Wrapper = styled.div`
  padding: 1em 0;
  width: 100%;
  height: auto;
  display: flex;
`;

const CustDetail = styled.div`
  width: 100%;
  height: auto;
  display: flex;
  flex-direction: column;
`;

const CustNamePlate = styled.p`
  font-size: 1.1em;
  padding: 0.5em 0.8em;
  background-color: #eaf2fa;
  border-radius: 0.3em;
  margin-right: 20px;
  color: #407fcc;
`;

const CustName = styled.p`
  font-size: 1.2em;
`;

const CustRole = styled.p`
  text-transform: capitalize;
  color: #666;
  font-size: 0.9em;
  padding-top: 2px;
`;

const Customer: React.FC<CustomerProps> = (props: CustomerProps) => {
  return (
    <Wrapper>
      <CustNamePlate>{props.customer.name.substring(0, 1)}</CustNamePlate>
      <CustDetail>
        <CustName>{props.customer.name}</CustName>
        <CustRole>{props.customer.role.toLowerCase()}</CustRole>
      </CustDetail>
    </Wrapper>
  );
};

export default Customer;
