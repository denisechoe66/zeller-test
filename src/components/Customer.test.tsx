import renderer from 'react-test-renderer';
import Customer from './Customer';
import { Person } from '../interfaces/person';

describe('<Customer />', () => {
  const customerDetail: Person = {
    name: 'David Miller',
    role: 'ADMIN',
    email: 'david@gmail.com',
    id: '73bae2af-4fa4-4023-8829-1034604e7590'
  };
  it('should render Customer', () => {
    const component = renderer.create(<Customer customer={customerDetail} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
