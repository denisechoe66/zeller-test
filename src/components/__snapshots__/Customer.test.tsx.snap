// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Customer /> should render Customer 1`] = `
<div
  className="sc-bdfBQB ifOoGt"
>
  <p
    className="sc-dlfnuX csvATz"
  >
    D
  </p>
  <div
    className="sc-gsTEea iAvDKs"
  >
    <p
      className="sc-hKgJUU ddhPps"
    >
      David Miller
    </p>
    <p
      className="sc-eCstlR kRXOWZ"
    >
      admin
    </p>
  </div>
</div>
`;
